### DISCUSSION TITLE
Are Mobile Push Notifications secure?

### DISCUSSION THREAD

**Discussion by Svenum** [dani-garcia/vaultwarden#4721] [General]
Doku: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-Mobile-Client-push-notification

I am thinking about to enable the Mobile Push notification, but I saw that you need some Keys/IDs from Bitwarden directly. What does these keys do? Are the Secrets send over the Bitwarden Server to my self hosted Vaultwarden?

Is it secure to do this?

**Comment by BlackDex** [comment]
Those keys are needed to allow a Self-Hosted instance to connect to Bitwarden's Push service. The same procedure is needed for the Bitwarden Self-Hosted instances.

It will only send uuid's since that is how items are identified, and those items are reloaded.

**Reply by Svenum** [reply]
ok, so bitwarden only gets to know the uuids of the items that get synced via this service, am I right?

**Reply by BlackDex** [reply]
Yes
